src/components/legend/attributes.js

```diff
@@ -74,6 +74,19 @@ module.exports = {
             'Sets the amount of vertical space (in px) between legend groups.'
         ].join(' ')
     },
+    entrywidth: {
+        valType: 'number',
+        min: 0,
+        editType: 'legend',
+        description: 'Sets the width (in px or fraction) of the legend.',
+    },
+    widthmode: {
```

Comment:
@alexcjohnson 
The `widthmode` may be confusing considering other `*width` attributes e.g. `itemwidth`.
What about renaming to `entrywidthmode`.
Alternatively those two new attribute could be inside an `entry` object:
```js
entry: { 
  width: ...,
  mode: ...
}
```

Reply:
that's a good point - also the other `*mode` attributes we have (`colorbar.lenmode`, `colorbar.thicknessmode`) match exactly the name of the value attribute plus `mode`. So yeah let's go for `entrywidthmode`.

Reply:
I renamed widthmode with entrywidthmode